ir/shared/src/main/scala/org/scalajs/ir/Names.scala

```diff
@@ -396,7 +396,8 @@ object Names {
     }
 
     def constructor(paramTypeRefs: List[TypeRef]): MethodName = {
-      new MethodName(ConstructorSimpleName, paramTypeRefs, VoidRef,
+      // @unchecked for the initialization checker of Scala 3
+      new MethodName(ConstructorSimpleName: @unchecked, paramTypeRefs, VoidRef,
```

Comment:
This is a bug in the checker, which is fixed in https://github.com/lampepfl/dotty/pull/15467.

Reply:
Ah good. I tried 3.2.0-RC1, but the fix came after even that.

Reply:
Indeed, when trying with the nightly build, the init checker was happy without this change. I have reverted it.